### PR TITLE
Updates docs to adoc, and licenses around

### PR DESCRIPTION
**Pull request by alessioborraccino**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+Template could be found here: link:https://github.com/d4l-data4life/hc-readme-template/blob/main/TEMPLATE_CHANGELOG.adoc[CHANGELOG]
+
+[discrete]
+==== Versions
+
+toc::[]
+
+== link:{link-repository}/releases/latest[Unreleased]
+
+See link:{link-repository}/compare/v1.16.0...main[changeset]
```
Comment (bitPogo) [inline CHANGELOG.adoc:31]: Is the version correct? - the lastest release states 0.22.1

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+=== Security
+
+=== Bumped
+
+=== Deprecated
+
+=== Migration
+
+== link:{link-repository}/releases/tag/v0.23.0[0.23.0]
```
Comment (bitPogo) [inline CHANGELOG.adoc:49]: Is the version correct? - the lastest release states 0.22.1

File: DEVELOPING.adoc [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+Default output format: json
+----
+
+NOTE: Access key ID and secret are secretly stored and must be asked for.
+
+== Build
+
+=== File generation
+
+STU3 models/tests and r4 protocol conformations/tests are autogenerated via the parser and sourcery.
```
Comment (bitPogo) [inline DEVELOPING.adoc:90]: Nitpick: R4, I suppose

File: DEVELOPING.adoc [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+* `release/{major}.{minor}.{patch})/prepare-{major}.{minor}.{patch}`
+
+=== Steps
+
+* Update link:CHANGELOG.adoc[CHANGELOG]
+* Update documentation if needed
+* Run prepare release script (eg. `bundle exec fastlane ios prepare_release version:"v1.0.0" api_token:"xxxxxxx"`)
+* On github, publish the release manually by setting the new tag (eg. `v1.0.0`)
+
+NOTE: Prepare release script will fail if there is no new version defined in the changelog
```
Comment (bitPogo) [inline DEVELOPING.adoc:132]: Nitpick: `The Prepare release script [...]`

File: DEVELOPING.adoc [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+=== Prepare release script steps
```
Comment (bitPogo) [inline DEVELOPING.adoc:134]: Maybe put this before `Steps` it is a bit confusing to read it in this order in my opinion.

File: DEVELOPING.adoc [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+The script executes the following commands:
+
+* Update new versions in the README.adoc and project files (including badges)
+* Do sanity checks (Proper version number in changelog etc.)
+* Generate documentation
+* Build framework and upload it to S3
+* Update Package swift info (url and checksum)
+* Commit and push changes
+* Create GitHub release draft with all of the information from link:CHANGELOG.adoc[CHANGELOG]
```
Comment (bitPogo) [inline DEVELOPING.adoc:144]: `Create a GitHub release draft with all of the information from the CHANGELOG`

File: DEVELOPING.adoc [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+[source, bash]
+----
+bundle exec fastlane ios prepare_release version:"1.0.0" api_token:"xxxxxxx"
+----
+
+=== Generate documentation
+
+Documentation is generated using Asciidoctor, all of the resoures can be found in link:asciidoc/[asciidoc], and generated HTML documentation can be found in link:docs/(docs).
```
Comment (bitPogo) [inline DEVELOPING.adoc:153]: Maybe call it just `docs` or something simply because ASCIIDoc is a format and not a common term.

Comment (bitPogo) [reply DEVELOPING.adoc:153]: Also `link:docs/(docs)` is not displayed correctly.

File: README.adoc [hunk 8]
```diff
@@ -0,0 +1,118 @@
+= Data4Life Core iOS SDK
+:library_version: 1.16.0
+:doctype: article
+:!showtitle:
+:toc: macro
+:toclevels: 2
+:toc-title:
+:icons: font
+:imagesdir: assets/images
+:link-repository: https://github.com/d4l-data4life/d4l-fhir-ios
```
Comment (bitPogo) [inline README.adoc:10]: This should be now on line 2.

File: README.adoc [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+=== Table Of Contents
+toc::[]
+
+== About The Project
+
+Minimal FHIR standard models and data types for iOS.
+
+[Swift-FHIR](https://github.com/smart-on-fhir/Swift-FHIR) was used as inspiration for this project
+
+[Apple-FHIR](https://github.com/apple/FHIRModels) R4 models were (tepmorarily) ported over.
```
Comment (bitPogo) [inline README.adoc:68]: Both are not display correctly I suppose.

File: README.adoc [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+        <a>
+            <img src="assets/images/badge-platform-support.svg" alt="Platform"/>
+        </a>
+        <a>
+            <img src="assets/images/badge-manager-support.svg" alt="Manager"/>
+        </a>
+        <a href="LICENSE">
+            <img src="assets/images/badge-license.svg" alt="License"/>
+        </a>
+    </p>
```
Comment (bitPogo) [inline README.adoc:54]: Please take a look into the latest iteration of the Header in the README. An example can be found [here](https://github.com/d4l-data4life/hc-data-donation-sdk-kmp/pull/31).

**Review by bitPogo — CHANGES_REQUESTED** [review]
Please see below.

**Review by jose-cuadros-data4life — APPROVED** [review]
Please consider @bitPogo comments, but apart of his comments I couldn't find any other issue

File: CHANGELOG.adoc (hunk 2 above)
Comment (alessioborraccino) [reply CHANGELOG.adoc:49]: yes its coming

File: CHANGELOG.adoc (hunk 1 above)
Comment (alessioborraccino) [reply CHANGELOG.adoc:31]: done

File: DEVELOPING.adoc (hunk 3 above)
Comment (alessioborraccino) [reply DEVELOPING.adoc:90]: 👍

File: DEVELOPING.adoc (hunk 4 above)
Comment (alessioborraccino) [reply DEVELOPING.adoc:132]: 👍

File: DEVELOPING.adoc (hunk 5 above)
Comment (alessioborraccino) [reply DEVELOPING.adoc:134]: Its more relevant to the manual steps just described, the script steps are more if you want to go in detail on it

File: DEVELOPING.adoc (hunk 7 above)
Comment (alessioborraccino) [reply DEVELOPING.adoc:153]: 👍

File: DEVELOPING.adoc (hunk 6 above)
Comment (alessioborraccino) [reply DEVELOPING.adoc:144]: 👍

File: README.adoc (hunk 8 above)
Comment (alessioborraccino) [reply README.adoc:10]: 👍

File: README.adoc (hunk 10 above)
Comment (alessioborraccino) [reply README.adoc:54]: 👍

File: CODE-OF-CONDUCT.adoc [hunk 11]
```diff
@@ -1,6 +1,25 @@
-# Contributor Covenant Code of Conduct
-
-## Our Pledge
+= Contributor Covenant Code of Conduct
+:link-repository: https://github.com/d4l-data4life/d4l-sdk-ios
```
Comment (bitPogo) [inline CODE-OF-CONDUCT.adoc:2]: d4l-fhir-ios

File: CONTRIBUTING.adoc [hunk 12]
```diff
@@ -0,0 +1,106 @@
+= Contributing
+:link-repository: https://github.com/d4l-data4life/d4l-sdk-ios
```
Comment (bitPogo) [inline CONTRIBUTING.adoc:2]: d4l-fhir-ios

File: CONTRIBUTING.adoc [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+
+==== Create Pull Request
+
+Please use our title pattern: `[{ticket id}] {type of change} {short description}`:
+
+* Optional: Add `ticket id` in brackets if you have any, otherwise leave it out.
+* `type of change`:
+- *Add* for new features or functionality
+- *Change* for changes in existing features or functionality
+- *Remove* | for removed features or functionality
```
Comment (bitPogo) [inline CONTRIBUTING.adoc:63]: - `|`

File: CONTRIBUTING.adoc [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+* [ ] Continuous Integration build passing
+* [ ] Cross platform testing done for all supported platforms
+* [ ] Documentation updated
+* [ ] Changelog updated
+
+[#_licensing]
+== Licensing
+
+Due to the private link:LICENSE[License] this repository is published under, external contributors need to sign a
+Contributor License Agreement. Please get in contact with mailto:&#109;o&#98;&#x69;&#x6c;&#x65;&#x40;&#100;&#x61;&#x74;a&#52;&#x6c;&#105;f&#101;&#46;&#99;a&#114;e[&#109;o&#98;&#x69;&#x6c;&#x65;&#x40;&#100;&#x61;&#x74;a&#52;&#x6c;&#105;f&#101;&#46;&#99;a&#114;e] before contributing to the project.
```
Comment (bitPogo) [inline CONTRIBUTING.adoc:97]: Is the mail to still valid?

**Review by bitPogo — CHANGES_REQUESTED** [review]
Still nitpicks left

**Review by bitPogo — APPROVED** [review]
LGTM

**Pull request merged**